docs/src/reconciling-local-changes.md

```diff
@@ -0,0 +1,234 @@
+# Reconciling changes made outside of TPA
+Any changes made to a TPA created cluster that are not performed by
+changing the TPA configuration will not be saved in `config.yml`. This
+means that your cluster will have changes that the TPA configuration
+won't be able to recreate.
+
+This page shows how configuration is managed with TPA and the preferred
+ways to make configuration changes. We then look at strategies to make,
+and reconcile, the results of making manual changes to the cluster.
+
+## Why might I need to make manual configuration changes?
+The most common scenario in which you may need to make configuration
+changes outside of TPA is if the operation you are performing is not
+supported by TPA. The two most common such operations are destructive
+changes, such as removing a node, and upgrading the major version of
+Postgres.
+
+### Destructive changes
+In general TPA will not remove previously deployed elements of a
+cluster, even if these are removed from `config.yml`. This sometimes
+surprises people because a strictly declarative system should always
+mutate the deployed artifacts to match the declaration. However, making
+destructive changes to production database can have serious consequences
+so it is something we have chosen not to support.
+
+### Major-version Postgres upgrades
+TPA does not yet provide an automated mechanism for performing major
+version upgrades of Postgres. Therefore if you need to perform an
+in-place upgrade on an existing cluster this must be performed using
+other tools such as pg_upgrade or [bdr_pg_upgrade](https://www.enterprisedb.com/docs/pgd/latest/upgrades/bdr_pg_upgrade/#bdr_pg_upgrade-command-line).
+
+## What can happen if changes are not reconciled?
+A general issue with unreconciled changes is that if you deploy a new
+cluster using your existing `config.yml`, or provide your `config.yml` to
+EDB Support in order to reproduce a problem, it will not match the
+original cluster. In addition, there is potential for operational
+problems should you wish to use TPA to manage that cluster in future.
+
```

Comment:
if you have made direct changes to postgresql configuration file(s), a subsequent tpaexec deploy will overwrite the configuration and you can potentially lose the configuration changes that you made.

Reply:
I think I covered that in the later section where I cover configuration files briefly.